### ISSUE TITLE
Layout Issue

### ISSUE THREAD

**Issue by nabeelazam** [kevalpatel2106/PasscodeView#3]
I have added gradle in my project and trying to use PinView. 

PinView in my activity is not being l
oaded properly. I tried to download and run the code same issue exists in Sample code.

![screenshot_20170803-194641](https://user-images.githubusercontent.com/16275/28927673-9d31ad48-7884-11e7-81cd-16a18b152798.png)

**Comment by dphans** [comment]
I've same problem, bro 👍

**Comment by TooLazyy** [comment]
I have the same issue on android 25+ api.
I have found that getLocalVisibleRect() in PasscodeView return wrong screen size.
Dont know how to fix that properly.
its NOT a complete solution, but you can add following code.

```
pinView.postDelayed(new Runnable() {
            @Override
            public void run() {
                //controll fingerbox, keyboard and indicator visibility here
               
                //
                pinView.invalidate();
                pinView.requestLayout();
            }
        }, 100);
```

Let me know if you know how to fix that better.

**Comment by plugie** [comment]
I have the same problem.

**Comment by masaibar** [comment]
I have the same problem, too. 🙏

**Comment by kevalpatel2106** [comment]
Fix in #8 . Fix will be available in next release.

**Comment by kevalpatel2106** [comment]
Version 1.2.1 is released with above fix.

**Comment by praveenb** [comment]
@kevalpatel2106,
I still see this issue, like Pinview moving out of the screen, when wrong pin entered, even with latest version 1.2.1. 
![pinview](https://user-images.githubusercontent.com/592596/38083381-08dbdaee-3367-11e8-824c-e4b426166667.png)

Please see attached image, Pinview moving to left side a little out of the screen for every wrong pin entry.
But above @TooLazyy solution code, fixed the issue. Hope it'll be fix in next version.

just FYI, I downloaded git source code and run it. I see its working fine i.e. Pinview not moving outside screen. I'm wondering why this same not working when integrating in a project with gradle library.

anyway, @kevalpatel2106, Thanks for the nice work.
Thanks @TooLazyy 
Thank you.

**Comment by kevalpatel2106** [comment]
@praveenb Can you share the file in which you made the changes suggested by @TooLazyy? As per my understanding since PR #8 `getLocalVisibleRect()` is not being used in the measuring phase. Also which branch you were using from the git source code?

**Comment by kevalpatel2106** [comment]
@praveenb Can you confirm the issue is still present in the dev branch also?

**Comment by praveenb** [comment]
@kevalpatel2106 , Since the Pinview moving out of screen, when Authentication failed. I applied the @TooLazyy solution code in onAuthenticationFailed() method in my project Activity.
Solved the issue.

I've downloaded master branch code, I see its working fine. I'll check dev branch soon and let you know.

@kevalpatel2106 ,
I tried dev branch, its crashing when loading screen itself with Pinview. But with pattern view its loaded fine. So not able to check moving out of screen issue with Pinview.

Thank you.
